### PR TITLE
fix(parse): count lines like ast/tokenize, not str.splitlines()

### PR DESCRIPTION
**Pull request by dmadisetti**
## 📝 Summary

On bulk examination of notebooks, I found edge-cases that we hard failed on parsing due to `str.splitlines()`.
This PR fixes the failure by specifically splitting lines on `\n` and `\r` + adds unit tests to prevent regression.

<a href="https://cubic.dev/pr/marimo-team/marimo/pull/9957?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Review by mscolnick — DISMISSED** [review]

File: tests/_ast/test_parse_control_chars.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+from marimo._ast.parse import parse_notebook
+
+# A form-feed (\x0c) inside an `mo.md` string. The tabs (\t, shown as `\times`
+# in the literal) come from an unescaped LaTeX `\times`. marimo generated this
+# file but, today, asserts out when parsing it back.
+NOTEBOOK_FORMFEED_ASSERT = 'import marimo\n__generated_with = "0.17.7"\napp = marimo.App(\n)\n@app.cell\ndef _(mo):\n    mo.md("""\n    P=\x0c    rac{TP}{TP+FP}\n    R=\x0c    rac{TP}{TP+FN}\n    F1 = \x0c    rac{2\times P\times R}{P+R}\n    """)\n    return\n\n@app.cell\ndef _(mo):\n    mo.md("""\n    """)\n    return\n\n@app.cell\ndef _():\n    pass\n    return\nif __name__ == "__main__":\n    app.run()\n'
+
+# Same root cause; parametrized `@app.cell(hide_code=True)` decorators make the
+# mis-extracted fragment untokenizable -> tokenize.TokenError instead.
```
Comment (dmadisetti) [inline tests/_ast/test_parse_control_chars.py]:
```suggestion
# incorrectly extracted fragment untokenizable -> tokenize.TokenError instead.
```
